osbs/build/build_request.py

```diff
@@ -260,31 +260,41 @@ def render_unique_tag_only(self):
 
         self.dj.remove_plugin('postbuild_plugins', 'tag_from_config')
 
-    def set_secret_for_plugin(self, plugin, secret):
-        has_plugin_conf = self.dj.dock_json_has_plugin_conf(plugin[0],
-                                                            plugin[1])
+    def set_secret_for_plugin(self, secrets, plugin=None):
+        has_plugin_conf = False
+        if plugin is not None:
+            has_plugin_conf = self.dj.dock_json_has_plugin_conf(plugin[0],
+                                                                plugin[1])
         if 'secrets' in self.template['spec']['strategy']['customStrategy']:
-            if has_plugin_conf:
-                secret_path = os.path.join(SECRETS_PATH, secret)
-                logger.info("Configuring %s secret at %s", secret, secret_path)
+            if not plugin or has_plugin_conf:
+                if not isinstance(secrets, dict):
+                    secrets = {secrets : ""}
```

Comment:
I'm not sure I like parameters that get automatically changed to a different type.

How about making the method signature `set_secret_for_plugin(self, secret, plugin=None, mount_path=None)`? Then iterating over the secrets in `token_secrets` could be done in render() instead of here.